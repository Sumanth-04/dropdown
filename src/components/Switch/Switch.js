import React from 'react';
import Switch from 'rc-switch';
import PropTypes from 'prop-types';
import './index.scss';
//import 'rc-switch/assets/index.css';

class SwitchTab extends React.Component {
  static propTypes = {
		className: PropTypes.string,
  };
  
  render() {
    return (
      <div>
        <Switch className={`rd-switch ${this.props.className}`}
			      {...this.props}
        />
      </div>
    );
  }
}

export default SwitchTab;