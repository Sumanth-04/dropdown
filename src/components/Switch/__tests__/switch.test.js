import React from "react";
import renderer from "react-test-renderer";
import Switch from "../../Switch/Switch";

it("Switch uses default scss and can have checkedChildren uncheckedChildren as props", () => {
  const tree = renderer
    .create(
      <Switch
        disabled={false}
        checkedChildren="开"
        unCheckedChildren="关"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
