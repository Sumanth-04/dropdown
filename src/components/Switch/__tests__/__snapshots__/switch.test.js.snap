// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch uses default scss and can have checkedChildren uncheckedChildren as props 1`] = `
<div>
  <button
    aria-checked={false}
    className="rc-switch rd-switch undefined"
    disabled={false}
    onClick={[Function]}
    onKeyDown={[Function]}
    role="switch"
    type="button"
  >
    <span
      className="rc-switch-inner"
    >
      å…³
    </span>
  </button>
</div>
`;
